lib/Doctrine/ORM/Query/TreeWalkerChain.php

```diff
@@ -33,6 +33,7 @@ class TreeWalkerChain implements TreeWalker
      * The tree walkers.
      *
      * @var TreeWalker[]
+     * @psalm-var TreeWalkerChainIterator
```

Comment:
:thinking: why not use `@var TreeWalkerChainIterator` here?

Reply:
To maintain auto-completion in PHPStorm. If we put in `@var`, each foreach loop will result in a mixed object.
It's become a reflex for me when I add psalm properties that will cause this kind of issue but I can adapt if needed.

Reply:
I've seen on Twitter that integration with Psalm and PHPStorm was coming and that it included using some annotations: https://twitter.com/psalmphp/status/1280848618895880197

Reply:
Doesn't look like it's there yet though: https://github.com/JetBrains/phpstorm-psalm-plugin/issues/16, so I suppose it's fine to have both.

Reply:
Yeah, they expect to include it in the EAP starting September. So it should be released as stable by the end of the year. And we will have to see if generics are included and work well. I'll gladly submit a PR then to remove excess information if everything goes well :)